apps/protailwind/src/video/lesson-list.tsx

```diff
@@ -140,12 +138,14 @@ export const LessonList: React.FC<{
                         Exercise
                       </Link>
                     </li>
-                    <SolutionLink
-                      module={module}
-                      exercise={exercise}
-                      section={section}
-                      path={path}
-                    />
+                    {exercise.solution != null && (
```

Comment:
I'm having a TS problem, and I'm probably thinking too hard about this, but I was trying to filter if the `exercise` has a `solution`. However, since the type comes from `import {type LessonResource} from '@skillrecordings/skill-lesson/schemas/lesson-resource'` and it doesn't seem to be able to access `solution` even tho its part of the object. 

I went down the road and added `solution` to the schema from `skill-lesson`, but this breaks builds for other apps, such as total typescript. 

Any ideas  how can access `solution` without breaking other apps?

Reply:
**all exercises have solutions** or they should be a different type. struggling with an application fix for what is a data problem.

Reply:
an exercise without a solution is an explainer!